server/core/src/main/java/io/whitefox/core/services/IcebergSharedTable.java

```diff
@@ -0,0 +1,42 @@
+package io.whitefox.core.services;
+
+import io.whitefox.core.Metadata;
+import io.whitefox.core.ReadTableRequest;
+import io.whitefox.core.ReadTableResultToBeSigned;
+import java.util.Optional;
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.iceberg.Table;
+
+public class IcebergSharedTable implements AbstractSharedTable {
+
+  private final Table icebergTable;
+  private final TableSchemaConverter tableSchemaConverter;
+
+  private IcebergSharedTable(Table icebergTable, TableSchemaConverter tableSchemaConverter) {
+    this.icebergTable = icebergTable;
+    this.tableSchemaConverter = tableSchemaConverter;
+  }
+
+  public static IcebergSharedTable of(
+      Table icebergTable, TableSchemaConverter tableSchemaConverter) {
+    return new IcebergSharedTable(icebergTable, tableSchemaConverter);
+  }
+
+  public static IcebergSharedTable of(Table icebergTable) {
+    return new IcebergSharedTable(icebergTable, new TableSchemaConverter());
+  }
+
+  public Optional<Metadata> getMetadata(Optional<String> startingTimestamp) {
+    throw new NotImplementedException();
+  }
+
+  @Override
+  public Optional<Long> getTableVersion(Optional<String> startingTimestamp) {
+    throw new NotImplementedException();
+  }
+
+  @Override
+  public ReadTableResultToBeSigned queryTable(ReadTableRequest readTableRequest) {
+    throw new NotImplementedException();
+  }
```

Comment:
is something missing here or this class should be deleted?

Reply:
Nope, these are the api's that we have to implement in the next issues:

getMetadataAPI: https://github.com/agile-lab-dev/whitefox/issues/180
getTableVersion: https://github.com/agile-lab-dev/whitefox/issues/181
queryTable: https://github.com/agile-lab-dev/whitefox/issues/182